mod/notify.php

```diff
@@ -47,38 +38,37 @@ function notify_init(&$a) {
 
 if(! function_exists('notify_content')) {
 function notify_content(&$a) {
-	if(! local_user())
-		return login();
-
-		$notif_tpl = get_markup_template('notifications.tpl');
+	if(! local_user()) return login();
 
-		$not_tpl = get_markup_template('notify.tpl');
-		require_once('include/bbcode.php');
+	$nm = new NotificationsManager();
+	
+	$notif_tpl = get_markup_template('notifications.tpl');
 
-		$r = q("SELECT * from notify where uid = %d and seen = 0 order by date desc",
-			intval(local_user())
-		);
+	$not_tpl = get_markup_template('notify.tpl');
+	require_once('include/bbcode.php');
 
-		if (count($r) > 0) {
-			foreach ($r as $it) {
-				$notif_content .= replace_macros($not_tpl,array(
-					'$item_link' => $a->get_baseurl(true).'/notify/view/'. $it['id'],
-					'$item_image' => $it['photo'],
-					'$item_text' => strip_tags(bbcode($it['msg'])),
-					'$item_when' => relative_date($it['date'])
-				));
-			}
-		} else {
-			$notif_content .= t('No more system notifications.');
+	$r = $nm->getAll(array('seen'=>0));
+	if ($r!==false && count($r) > 0) {
+		foreach ($r as $it) {
+			$notif_content .= replace_macros($not_tpl,array(
+				'$item_link' => $a->get_baseurl(true).'/notify/view/'. $it['id'],
```

Comment:
just a hint:
we can now use `app::get_baseurl()` instead of `$a->get_baseurl()`  so we don't have to transmit `$a` when the only thing we need from `$a` would be the baseurl